diff --git a/src/specs/NativeRNMBXLocationModule.ts b/src/specs/NativeRNMBXLocationModule.ts
index 9b36cebfeac7d2c91720e08a67d89e77d9988192..0130aa354721799393bc2d3357595fa6a94b8d0b 100644
--- a/src/specs/NativeRNMBXLocationModule.ts
+++ b/src/specs/NativeRNMBXLocationModule.ts
@@ -1,6 +1,4 @@
 import { type TurboModule, TurboModuleRegistry } from 'react-native';
-// @ts-ignore - CI environment type resolution issue for CodegenTypes
-import type { EventEmitter } from 'react-native/Libraries/Types/CodegenTypes';
 
 type LocationEvent = {
   type: string; //"userlocationdupdated"
@@ -26,7 +24,7 @@ export interface Spec extends TurboModule {
   simulateHeading(changesPerSecond: number, increment: number): void;
   setLocationEventThrottle(throttle: number): void;
 
-  readonly onLocationUpdate: EventEmitter<LocationEvent>;
+  onLocationUpdate(callback: (event: LocationEvent) => void): void;
 }
 
 export default TurboModuleRegistry.getEnforcing<Spec>('RNMBXLocationModule');
